presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py

```diff
@@ -0,0 +1,333 @@
+
+        return recognizers
+
+    default_values = {
+        "supported_languages": ["en"],
+        "recognizers": [],
+        "global_regex_flags": re.DOTALL | re.MULTILINE | re.IGNORECASE,
+    }
+
+    def create_recognizer_registry(self) -> RecognizerRegistry:
+        """Create a recognizer registry according to configuration loaded previously."""
+        fields = {
+            "supported_languages": None,
+            "recognizers": None,
+            "global_regex_flags": None,
+        }
+
+        for field in fields.keys():
+            if field not in self.configuration:
+                logger.warning(
+                    f"{field} not present in configuration, "
+                    f"using default value instead: {self.default_values[field]}"
+                )
+            fields[field] = self.configuration.get(field, self.default_values[field])
+
+        self.supported_languages = fields["supported_languages"]
+
+        recognizer_instances = []
+        predefined, custom = self._split_recognizers(fields["recognizers"])
+        for recognizer_conf in predefined:
+            for language_conf in self._get_recognizer_languages(recognizer_conf):
+                if self._is_recognizer_enabled(recognizer_conf):
+                    copied_recognizer_conf = {
+                        k: v
+                        for k, v in self._get_recognizer_items(recognizer_conf)
+                        if k not in ["enabled", "type", "supported_languages", "name"]
+                    }
+                    kwargs = {**copied_recognizer_conf, **language_conf}
+                    recognizer_name = self._get_recognizer_name(recognizer_conf)
+                    recognizer_cls = self._get_existing_recognizer_cls_by_name(
+                        recognizer_name
+                    )
+                    recognizer_instances.append(recognizer_cls(**kwargs))
+
+        for recognizer_conf in custom:
+            if self._is_recognizer_enabled(recognizer_conf):
+                recognizer_instances.extend(
+                    self._create_custom_recognizers(recognizer_conf)
+                )
+
+        for recognizer_conf in recognizer_instances:
+            if isinstance(recognizer_conf, PatternRecognizer):
+                recognizer_conf.global_regex_flags = fields["global_regex_flags"]
+
+        fields["recognizers"] = recognizer_instances
+
+        return RecognizerRegistry(**fields)
+
+    @staticmethod
+    def _get_full_conf_path(
+        default_conf_file: Union[Path, str] = "default_recognizers.yaml"
+    ) -> Path:
+        """Return a Path to the default conf file."""
+        return Path(Path(__file__).parent, "../conf", default_conf_file)
+
+    @staticmethod
+    def _get_all_existing_recognizers(
+        cls: Optional[Type[EntityRecognizer]] = None,
+    ) -> Set[Type[EntityRecognizer]]:
+        """
+        Return all subclasses of EntityRecognizer, recursively.
```

Comment:
New method to collect all recognizers which exist in the codebase

Reply:
That's pretty, I'd move it to recognizer_registry_utils.py now that we have a dedicated folder :D